src/cluster_copier.cc

```diff
@@ -116,31 +106,33 @@ namespace grpc_labview {
     //---------------------------------------------------------------------
     void ClusterDataCopier::CopyFromCluster(LVMessage& message, int8_t* cluster)
     {
-        message._values.clear();
-        std::map<std::string, int> oneof_containerToSelectedIndexMap; // Needed to serialize only the field related to the selected_index
-        for (auto val : message._metadata->_mappedElements)
+        message.Clear();
+        for (auto& fieldMetadata : message._metadata->_elements)
         {
-            auto fieldMetadata = val.second;
-            if (fieldMetadata->isInOneof)
+            if (fieldMetadata->isInOneof && fieldMetadata->protobufIndex < 0)
             {
-                if (fieldMetadata->protobufIndex < 0)
-                {
-                    // set the map of the selected index for the "oneofContainer" to this protobuf Index
-                    assert(oneof_containerToSelectedIndexMap.find(fieldMetadata->oneofContainerName) == oneof_containerToSelectedIndexMap.end());
-                    auto selected_index = *(int*)(cluster + fieldMetadata->clusterOffset);
-                    oneof_containerToSelectedIndexMap.insert(std::pair<std::string, int>(fieldMetadata->oneofContainerName, selected_index));
-                }
+                // set the map of the selected index for the "oneofContainer" to this protobuf Index
+                assert(message._oneofContainerToSelectedIndexMap.find(fieldMetadata->oneofContainerName) == message._oneofContainerToSelectedIndexMap.end());
+                auto selected_index = *(int*)(cluster + fieldMetadata->clusterOffset);
+                message._oneofContainerToSelectedIndexMap.insert(std::pair<std::string, int>(fieldMetadata->oneofContainerName, selected_index));
```

Comment:
nit: You should be able to use an initializer list here instead of spelling out the type.

```suggestion
                message._oneofContainerToSelectedIndexMap.insert({ fieldMetadata->oneofContainerName, selected_index });
```